contracts/ExampleNFT.cdc

```diff
@@ -117,26 +117,23 @@ access(all) contract ExampleNFT: NonFungibleToken {
                     let fooTraitRarity = MetadataViews.Rarity(score: 10.0, max: 100.0, description: "Common")
                     let fooTrait = MetadataViews.Trait(name: "foo", value: self.metadata["foo"], displayType: nil, rarity: fooTraitRarity)
                     traitsView.addTrait(fooTrait)
-                    
+
                     return traitsView
             }
             return nil
         }
     }
 
-    access(all) resource Collection: NonFungibleToken.Collection {
+    // Deprecatred: Only here for backward compatibility.
+    access(all) resource interface ExampleNFTCollectionPublic {}
+
+    access(all) resource Collection: NonFungibleToken.Collection, ExampleNFTCollectionPublic {
         /// dictionary of NFT conforming tokens
         /// NFT is a resource type with an `UInt64` ID field
-        access(contract) var ownedNFTs: @{UInt64: ExampleNFT.NFT}
-
-        access(all) var storagePath: StoragePath
-        access(all) var publicPath: PublicPath
+        access(contract) var ownedNFTs: @{UInt64: {NonFungibleToken.NFT}}
```

Comment:
Yeah I understand the concerns.

The reason for not allowing changing the field is, if the type was changed in the contract, then all of a sudden the existing data stored no longer match the type of this field. So in order make it compatible, we would need to also migrate the values to the changed type, by running a migration. However, this is going to be very complex because:
1) We would have to keep track of all the field types changes, and then during the migration, we would need to backtrack and find the type of the field, and then change the type of value accordingly.
2) Some changes to types may not be compatible, but we can only know that for sure during the migration of the value. For e.g: given the type of the map's values were previously `NonFungibleToken.NFT`, it is theoretically possible to have any `NonFungibleToken.NFT` in the map, and not just `ExampleNFT.NFT` (maybe not in this particular example, but possible in other similar cases). i.e: This is "narrowing" the type, and narrowing isn't always guaranteed to be succeed. Where as broadening the type (changing from `ExampleNFT.NFT` to `NonFungibleToken.NFT`) would. So given that the contract is already staged by the time we run the migrations, if we run into such incompatibilities, those data would be forever corrupted. So that's why the contract update validator only allows changes that are 100% guaranteed to be valid.

Again, I get this may not be ideal, maybe we could do some brainstorming and try to find a middle ground.

Reply:
okay that is fine for now. I mostly want it there as an example for people who are creating new collections, but I guess we can just change it after the cadence 1.0 upgrade is complete